### PR TITLE
adding lr_decay_steps and refactoring get_scheduler

### LINKED ISSUE #61: Add a decay option to `get_scheduler` and refactor to use SequentialLR
As a follow-up to #57, we should refactor the `get_scheduler()` function to make it possible to add a linear warm-up and decay to any scheduler. The [SequentialLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.SequentialLR.html#sequentiallr) scheduler should be able to help with this, and should make the function simpler as well.

### PR DESCRIPTION
**Pull request by chanind**
This PR adds a `lr_decay_steps` config option which will linearly decay learning rate to 0 at the end of training. This PR simplifies the `get_scheduler` function by removing `constantwithwarmup`, `linearwarmupdecay`, and `cosineannealingwarmup` options. Instead, any scheduler can now have a linear warmup and linear decay by setting the `lr_warm_up_steps` and `lr_decay_steps`.

This also required changing some of the options passed to `get_scheduler`. This function now takes in `lr` as well, since there's a discrepancy with how torch's cosine annealing scheduler works compared to how the other schedulers. Specifically, it seems that torch's cosine annealing actually sets the final LR to the value specified by eta_min, but the other schedulers don't know what the current learning rate is and instead can only set the LR to a portion of its original value. In order to address this discrepancy, we need to know the LR so we can calculate what the expected end LR should be as a portion of the base LR. This also has a nice side-effect of meaning that the `lr_end` value is the actual final LR, instead of a portion of the original LR which felt unexpected.

closes #61

### CODE REVIEW

**Comment by jbloomAus** [comment]
Looks great! thanks!
